openwisp_controller/config/admin.py

```diff
@@ -926,6 +926,16 @@ def __init__(self, *args, **kwargs):
                 kwargs['initial'].update({'backend': app_settings.DEFAULT_VPN_BACKEND})
             super().__init__(*args, **kwargs)
 
+        def clean(self):
+            cleaned_data = super().clean()
+            backend = cleaned_data.get('backend')
+            auth_token = cleaned_data.get('auth_token')
+            if backend == app_settings.VPN_BACKENDS[3][0] and not auth_token:
+                self.add_error(
+                    'auth_token', _('Auth token is required for this VPN backend.')
+                )
+            return cleaned_data
```

Comment:
This validation should exist in the model itself. The previous implementation in my last review was correct. 

I only asked to add the `required` attribute in the HTML. Since, you mentioned that it has no affect in the submission of the form. 

Can you please move the validation back to the model?

Reply:
Okay, I'll move that back to `vpn.py` :+1: